### PR TITLE
[NEW PORT]: sysutils/alloy

### PR DESCRIPTION
**Pull request by zalegrala**
- [x] Tested with poudriere
- [x] Runs on 14.0

### CODE REVIEW

File: sysutils/alloy/files/alloy.in [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+: ${alloy_group:="nobody"}
+: ${alloy_args:=""}
+: ${alloy_storage_path:="/var/alloy"}
+: ${alloy_listen_address:=":9200"}
+
+pidfile=/var/run/alloy.pid
+command="/usr/sbin/daemon"
+procname="%%PREFIX%%/bin/alloy"
+run="run /usr/local/etc/alloy.flow"
+command_args="-S -R 3 -T ${name} -P ${pidfile} /usr/bin/env ${procname} ${run} \
```
Comment (zalegrala) [inline sysutils/alloy/files/alloy.in:41]: This works but I would like guidance on the right combination of supervisor and restart settings.  It seems that if we write the supervisor pid with `-P` the status of the daemon cannot be determined with a `service alloy status`.  Additionally, if we only write the child pid with `-p`, then we cannot stop the daemon, since `daemon` will restart it.  What is the right approach here?

File: sysutils/alloy/Makefile [hunk 2]
```diff
@@ -0,0 +1,49 @@
+PORTNAME=	alloy
+PORTVERSION=	1.0.0
+PORTREVISION=	7
+DISTVERSIONPREFIX=v
+CATEGORIES=	sysutils
+
+MAINTAINER=	contact@zach.fi
```
Comment (zalegrala) [inline sysutils/alloy/Makefile:7]: I'm happy to be the contact, but I have no interest in maintaining freebsd accounts.

**Comment by mfechner** [comment]
I currently test build this MR and will commit it if it succeeds (I did some small cosmetic changes to it), you maybe want to close this MR by yourself after it is commited.

**Comment by mfechner** [comment]
Thanks a lot for your contribution @zalegrala

File: sysutils/alloy/Makefile [hunk 3]
```diff
@@ -0,0 +1,49 @@
+PORTNAME=	alloy
+PORTVERSION=	1.0.0
+PORTREVISION=	1
```
Comment (lwhsu) [inline sysutils/alloy/Makefile:3]: Why adding a new port with `PORTREVISION` set? It seems not really necessary.

File: sysutils/alloy/pkg-descr [hunk 4]
```diff
@@ -0,0 +1,3 @@
+OpenTelemetry Collector distribution with programmable pipelines
+
+WWW: https://github.com/grafana/alloy
```
Comment (rosorio) [inline sysutils/alloy/pkg-descr:3]: Has to be moved to Makefile [handbook](https://docs.freebsd.org/en/books/porters-handbook/book/#makefile-www)

File: sysutils/alloy/pkg-descr [hunk 5]
```diff
@@ -0,0 +1,3 @@
+OpenTelemetry Collector distribution with programmable pipelines
```
Comment (rosorio) [inline sysutils/alloy/pkg-descr:1]: This is probably too short

File: sysutils/alloy/files/alloy.in (hunk 1 above)
Comment (rosorio) [reply sysutils/alloy/files/alloy.in:41]: I tend to think that the -R flag in services is not a good idea, we have other tools to ensure service restart after a crash. One example is [sysutils/daedalus](http://www.freshports.org/sysutils/daedalus)

File: sysutils/alloy/Makefile [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+GH_PROJECT=	alloy
+
+GO_PKGNAME=	github.com/${GH_ACCOUNT}/${GH_PROJECT}
+GO_MODULE=	github.com/${GH_ACCOUNT}/${GH_PROJECT}
+GO_TARGET=	github.com/${GH_ACCOUNT}/${GH_PROJECT}
+GO_BUILDFLAGS=	-ldflags='-X github.com/grafana/alloy/internal/build.Version=v${PORTVERSION}'
+
+USE_RC_SUBR=	alloy
+
+ # Copy the go.mod so we can get all the dist files during fetch.
```
Comment (rosorio) [inline sysutils/alloy/Makefile:27]: This looks wrong to me, You are not supposed to use internal variables of the Mk scripts, mostly because this can fail at any moment if the scripts are changed. The same way you are not supposed to use the DISTFILE as a temporary storage.

But, why is this required during thefetch process ?

File: sysutils/alloy/Makefile (hunk 3 above)
Comment (zalegrala) [reply sysutils/alloy/Makefile:3]: Sure, this was part of the process of testing, and reset to 1 when I had something working for this PR.  I'll drop.

File: sysutils/alloy/files/alloy.in (hunk 1 above)
Comment (zalegrala) [reply sysutils/alloy/files/alloy.in:41]: You are suggesting just drop -R but keep daemon, correct?

File: sysutils/alloy/Makefile (hunk 6 above)
Comment (zalegrala) [reply sysutils/alloy/Makefile:27]: The fetch process did not handle this go.mod correctly without this change.  This does the fetching of that go.mod from the source so the rest of the dependencies can be read.  Note that the single source tar contains the main program, and this sub module.

If there is another way to do this, please advise.  Feel free to test without this block and it will present itself.

Comment (zalegrala) [reply sysutils/alloy/Makefile:27]: Also note this line: https://github.com/grafana/alloy/blob/main/go.mod#L791

**Comment by mfechner** [comment]
> I've rebased and updated the description as well as the daemon flags. Is it better to close this and open a new PR? I appreciate the attention and feedback.

yes please provide a new one, as most of the changes are already in the commit I did.
Maybe have a look into Mk/Uses/go.mk on option GO_MOD_DIST, maybe that help you too..

File: sysutils/alloy/Makefile (hunk 6 above)
Comment (zalegrala) [reply sysutils/alloy/Makefile]: Starting to update the port for  the new version, there is another internal go mod replace.
```
===> Fetching github.com/grafana/alloy dependencies
go: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogreceiver@v0.0.0-00010101000000-000000000000 (replaced by ./internal/etc/datadogreceiver): reading internal/etc/datadogreceiver/go.mod: open /usr/ports/distfiles/go/sysutils_alloy/grafana-alloy-v1.2.1_GH0/internal/etc/datadogreceiver/go.mod: no such file or directory
```

**Comment by zalegrala** [comment]
Thank you @mfechner.  I've opened a [new PR](https://github.com/freebsd/freebsd-ports/pull/286) and [bug](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=280446) for the update.

File: sysutils/alloy/files/alloy.in (hunk 1 above)
Comment (rosorio) [reply sysutils/alloy/files/alloy.in]: Yes. You can keep daemon call if the application has no demonize mechanism (many ports do the same thing), but removing the -R helps to keep the things simpler.

After that, using the  -p switch to recover child pidfile,  the stop and status will start working as expected .
